### ISSUE TITLE
`unsorted-imports` (`I001`) - does not detect duplicated imports when one of the imports is inside a `TYPE_CHECKING` block

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#9985]
```py
from collections.abc import Iterable
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from collections.abc import Iterable  # no error
```
[playground](https://play.ruff.rs/8a663307-3871-4671-8ca1-3b7166634195)

usually if there are duplicated imports, `I001` detects it:
```py
from collections.abc import Iterable
from collections.abc import Iterable  # error: Import block is un-sorted or un-formatted
```
